File: 1288-remove-covered-intervals/solution.cpp
#include <vector>
#include <algorithm>

class Solution {
public:
    int removeCoveredIntervals(std::vector<std::vector<int>>& intervals) {
        // Sort: ascending start, descending end for same starts
        std::sort(intervals.begin(), intervals.end(), [](const std::vector<int>& a, const std::vector<int>& b) {
            if (a[0] == b[0]) {
                return a[1] > b[1];
            }
            return a[0] < b[0];
        });

        int count = 0;
        int max_r = 0;

        for (const auto& interval : intervals) {
            // If current interval's end is within the max_r, it's covered
            if (interval[1] > max_r) {
                // Not covered, update max_r and count it
                max_r = interval[1];
                count++;
            }
        }

        return count;
    }
};

File: 1288-remove-covered-intervals/solution.py
from typing import List

class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        # Sort: ascending start, descending end for same starts
        # This puts wider intervals first if they share a start point
        intervals.sort(key=lambda x: (x[0], -x[1]))
        
        count = 0
        max_r = 0
        
        for _, r in intervals:
            # If current r is greater than the farthest end point seen so far,
            # this interval is not covered.
            if r > max_r:
                max_r = r
                count += 1
                
        return count